Cataclysm/Druid.lua

```diff
@@ -662,39 +696,34 @@ spec:RegisterStateExpr("calc_bite_dpe", function()
     local avg_base_damage = 377.877920772
     local scaling_per_combo_point = 0.125
     local dmg_per_combo_point = 576.189844175
+    local cp = combo_points.current
 
     local base_cost = (buff.clearcasting.up and 0) or (25 * ((buff.berserk.up and 0.5) or 1))
     local excess_energy = min(25, energy.current - base_cost) or 0
 
-    local bite_damage =         avg_base_damage
-    bite_damage = bite_damage + dmg_per_combo_point * combo_points.current 
-    bite_damage = bite_damage + state.stat.attack_power * scaling_per_combo_point * combo_points.current
+    local bonus_crit = rend_and_tear_mod_shred
+    local damage_multiplier = (1 + talent.feral_aggression.rank*0.05) * (1 + excess_energy/25)
 
-    -- Hekili:Debug("bite_damage pre dmgc (%.2f), attPower (%d)", bite_damage, state.stat.attack_power)
-    -- bite_damage = calculate_damage(1, bite_damage, false, false, false, false) TODO fix
-    Hekili:Debug("bite_damage (%.2f), excess_energy (%d)", bite_damage, excess_energy)
+    local bite_damage = avg_base_damage + cp*(dmg_per_combo_point + state.stat.attack_power*scaling_per_combo_point)
 
-    bite_damage = bite_damage * (1 + excess_energy/25)
-    bite_damage = bite_damage * (1 + talent.feral_aggression.rank * 0.05)
-    
-    Hekili:Debug("bite_damage final (%.2f), excess_energy (%d)", bite_damage, excess_energy)
+    bite_damage = calculate_damage(bite_damage, 0, false, false, true, damage_multiplier, bonus_crit) 
+    Hekili:Debug("bite_damage (%.2f), excess_energy (%d)", bite_damage, excess_energy)
 
-    local bite_dpe = bite_damage / (base_cost + excess_energy)
+    local bite_dpe = bite_damage / (base_cost + excess_energy) -- TODO: check if this should include excess energy
```

Comment:
yes it should - cause we're calculating the damage using the excess energy and we also spend that energy. So omitting it in the dpe calc would falsen the result

Reply:
Yes, I think the same, my doubt arises from the fact that WoWSims does not include it or I didn't see it 🤔

Reply:
I think my original implementation was right, we only spend more than 25 energy if we have the glyph active. It only consumed 25 more, so between 25 and 50 with glyph - but without its always fixed 25.

Thats why the sim is only calculating with 25 all the time:
![grafik](https://github.com/Hekili/hekili/assets/126069971/d9e5fd89-37d7-411a-8aef-4033870c95c1)

Also matches the spend resource (-restored energy due to misses) in this sim result:
![grafik](https://github.com/Hekili/hekili/assets/126069971/e2ba9e46-1b06-490b-8a73-6200d6ead41c)

Reply:
so we should keep "+ excess energy" but it can only greater than 0 with the glyph active

Reply:
I could be wrong, but there isn't a glyph that alters the energy consumption of _Ferocious Bite_ in Cataclysm. The glyph allows you to heal for 1%. I don't use the glyph in the game, and it consumes from 25 to 50 energy if there's enough available.

![image](https://github.com/Hekili/hekili/assets/35671015/4f6b9313-9c4c-4bae-8a13-94e4b878cb9b)
[glyph WoWHead](https://www.wowhead.com/cata/spell=67598/glyph-of-ferocious-bite)

![image](https://github.com/Hekili/hekili/assets/35671015/9c974e3f-da2f-4fca-9038-ed2cd563c6a1)
[Ferocious Bite WoWHead](https://www.wowhead.com/cata/spell=22568/ferocious-bite)